### DISCUSSION TITLE
Data-Importer error while importing transfer

### DISCUSSION THREAD

**Discussion by taker218** [firefly-iii/firefly-iii#5731] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I'm using the spectre API and the data importer to import the transactions from my bank and so far it all works really good.
The only problem I have is the following:

The import fails on those transactions where it is a transfer between two of my asset accounts. I regularly have transactions between those accounts. One is my normal bank account (default asset account), the other one a savings account for stuff that get's paid once a year like insurance.

### Expected behaviour

Create a transfer between the two asset accounts

### Debug information

Debug information generated at 2022-02-07 11:39:08 UTC for Firefly III version **5.6.14**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.6.14 |
| Firefly III API | 1.5.5 |
| Build | 96, 05-02-2022 10:01:49 CET |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | cca859af-87a2-49a6-8a4f-3bcadfc1f92f |
| Using docker? | true |
| Layout |  |
| System TZ |  |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | info, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| DB version | 18 (exp. 18) |
| Login provider | eloquent  |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| Session start | 2022-02-01 00:00:00 |
| Session end | 2022-02-28 23:59:59 |
| Session first | 2021-11-30 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:96.0) Gecko/20100101 Firefox/96.0 |

Debug information generated at 2022-02-07 12:40:25 Europe/Amsterdam for Firefly III Data Importer version **0.8.0**.

| Scope | Version |
| --- | --- |
| FIDI | 0.8.0 |
| Build | 93, 06-02-2022 09:21:56 CET |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ |  |
| Browser TZ | Europe/Berlin |
| App environment | production |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:96.0) Gecko/20100101 Firefox/96.0 |

### Additional info

The error I get in the web UI is the following:
(2)transactions.0.destination_id: Could not find a valid destination account when searching for ID "192" or name "". (original value: "192")
(2)transactions.0.destination_name: Could not find a valid destination account when searching for ID "192" or name "". (original value: "(not found)")

**Comment by JC5** [comment]
What kind of an account is nr 192 in your installation?

**Reply by taker218** [reply]
It's set up as a savings account.

**Reply by JC5** [reply]
Then it *should* work. Assuming the other end is also an asset account. Do you have the original transaction from the logs?

**Reply by taker218** [reply]
Sure, I can send you the log file from the import.
where should I send it to?

**Reply by JC5** [reply]
Send to james@firefly-iii.org, thanks!

**Comment by JC5** [comment, marked as answer]
Thanks for the patience. This is a bug, and will be fixed in the next release of Firefly III. You may run into duplicate transactions, due to slightly changed import code, so remember to import with care.

**Reply by taker218** [reply]
Just had a new transfer and now it's working as it should :)
Thanks for the fix!
